main.py

```diff
@@ -322,6 +364,54 @@ def getNeighbours(self):
                 neighbours.append(myGame.getState())
         return neighbours
 
+class Button:
+    """
+    TODO
+    """
+    def __init__(self, position, text, oneClickFunction):
+        """
+        TODO
+        """
+        self.position = position
+        self.textPosition = (self.position[0]+30, self.position[1] +self.position[3]-28)
+        self.text = text
+        self.oneClickFunction = oneClickFunction
+        self.state = BUTTON_STATE_DEFAULT
+
+    def draw(self):
+        """
+        Function Button.draw(self):
+        Simple draw function that places the rectangle on a new canvas frame.
+        If the Button object is hovered or clicked it changes its appearance.
+        """
+        pygame.draw.rect(g_screen, BUTTON_COLOURS[self.state], self.position)
+        img = g_font.render(str(self.text), True, BLACK)
+        g_screen.blit(img, self.textPosition)
+
+    def checkHover(self, position):
+        """
+        TODO
+        """
+        if  position[0] > (self.position[0])\
+        and position[0] < (self.position[0] + self.position[2])\
+        and position[1] > (self.position[1])\
+        and position[1] < (self.position[1] + self.position[3]):
+            self.state = BUTTON_STATE_HOVERED
+        else:
+            self.state = BUTTON_STATE_DEFAULT
+
+    def checkClicked(self, position):
+        """
+        Function Button.clicked(self, position):
+        Determines if the clicked-event is used on a button, by comparing the mouse
+        position to the position and the surface of each button used in the game.
+
+        param[in]   position    current position of the mouse during a click-event
+        """
+        if self.state == BUTTON_STATE_HOVERED:
+            self.state = BUTTON_STATE_CLICKED
+            self.oneClickFunction()
+
```

Comment:
Könnte man direkt in ein eigenes File auslagern.

Reply:
In gameEngineElements.py ausgelagert.